Assets/AnkrSDK/Runtime/WalletConnect2/WalletConnect2.cs

```diff
@@ -0,0 +1,340 @@
+		}
+
+		public async UniTask<TResponseData> Send<TRequestData, TResponseData>(TRequestData data)
+			where TRequestData : Identifiable 
+			where TResponseData : IErrorHolder
+		{
+			CheckIfSessionCreated();
+
+			var topic = _sessionData.Value.Topic; 
+			var result = await _signClient.Request<TRequestData, TResponseData>(topic, data).AsUniTask();
+			return result;
+		}
+
+		public async UniTask<string> EthSign(string address, string message)
+		{
+			CheckIfSessionCreated();
+			if (!message.IsHex())
+			{
+				var rawMessage = Encoding.UTF8.GetBytes(message);
+
+				var byteList = new List<byte>();
+				var bytePrefix = "0x19".HexToByteArray();
+				var textBytePrefix = Encoding.UTF8.GetBytes("Ethereum Signed Message:\n" + rawMessage.Length);
+
+				byteList.AddRange(bytePrefix);
+				byteList.AddRange(textBytePrefix);
+				byteList.AddRange(rawMessage);
+
+				var hash = new Sha3Keccack().CalculateHash(byteList.ToArray());
+
+				message = "0x" + hash.ToHex();
+			}
+			
+			Debug.Log(message);
+
+			var request = new EthSignRequestData(address, message);
+			
+			var topic = _sessionData.Value.Topic;
+			var response = await _signClient.Request<EthSignRequestData, EthResponseData>(topic, request);
+			return response.Result;
+		}
+
+		public UniTask<string> EthPersonalSign(string address, string message)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> EthSignTypedData<T>(string address, T data, EIP712Domain eip712Domain)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> EthSendTransaction(params TransactionData[] transaction)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> EthSignTransaction(params TransactionData[] transaction)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> EthSendRawTransaction(string data, Encoding messageEncoding = null)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> WalletAddEthChain(EthChainData chainData)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> WalletSwitchEthChain(EthChain chainData)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public UniTask<string> WalletUpdateEthChain(EthUpdateChainData chainData)
+		{
+			throw new System.NotImplementedException();
+		}
+
+		public async void Dispose()
+		{
+			await Disconnect();
+		}
+
+		public UniTask Quit()
+		{
+			return Disconnect();
+		}
+
+		public async UniTask OnApplicationPause(bool pauseStatus)
+		{
+			if (Status == WalletConnect2Status.Uninitialized)
+			{
+				throw new InvalidOperationException("WalletConnect is not initialized");
+			}
+
+			if (pauseStatus)
+			{
+				await Disconnect();
+			}
+			else
+			{
+				await Connect();
+			}
+		}
+		
+		private async UniTask Disconnect()
+		{
+			if (_signClient == null)
+			{
+				return;
+			}
+
+			if (ConnectionPending)
+			{
+				return;
```

Comment:
I think we should also ensure that if the current state is not Disconnected we still invoke SessionStatusUpdated